checks/junit.go

```diff
@@ -70,22 +77,49 @@ func (c *JunitChecker) Check(extConfig external.Check) *pkg.CheckResult {
 	if junitCheck.GetDisplayTemplate() != "" {
 		textResults = true
 	}
+	interval := junitCheck.GetInterval()
+	timeout := junitCheck.GetTimeout()
 	var junitStatus JunitStatus
 	template := junitCheck.GetDisplayTemplate()
 	pod := &corev1.Pod{}
 	pod.APIVersion = corev1.SchemeGroupVersion.Version
 	pod.Kind = podKind
+	pod.Labels = map[string]string{
+		junitCheckSelector: junitCheckLabelValue,
+	}
 	if junitCheck.GetNamespace() != "" {
 		pod.Namespace = junitCheck.GetNamespace()
 	} else {
 		pod.Namespace = corev1.NamespaceDefault
 	}
 	if junitCheck.GetName() != "" {
-		pod.Name = junitCheck.GetName()
+		pod.Name = junitCheck.GetName() + "-" + strconv.Itoa(int(start.Unix()))
 	} else {
 		name := rand.String(5)
 		pod.Name = strings.ToLower(name)
 	}
+	existingPods := getJunitPods(c.kommons, pod.Namespace)
+	if len(existingPods) != 0 {
+		for _, junitPod := range existingPods {
+			obj, err := c.kommons.GetByKind(podKind, junitPod.Namespace, junitPod.Name)
+			if err != nil {
+				return junitFailF(junitCheck, textResults, junitStatus, template, "error fetching the pod: %v", err)
+			}
+			if obj != nil {
+				logger.Tracef("pod already exist")
+				createTime := obj.GetCreationTimestamp()
+				duration := time.Since(createTime.Time)
+				dur := uint64(duration.Seconds())
```

Comment:
Don't use abbreviations except for those that are extremely common

In this case I don't think you need the extra objects

```
time.Since(createTime.Time) > time.Seconds(interval * 2)

```